Microsoft.Toolkit.Mvvm/DependencyInjection/Ioc.cs

```diff
@@ -0,0 +1,148 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using Microsoft.Extensions.DependencyInjection;
+
+#nullable enable
+
+namespace Microsoft.Toolkit.Mvvm.DependencyInjection
+{
+    /// <summary>
+    /// A type that facilitates the use of the <see cref="IServiceProvider"/> type.
+    /// The <see cref="Ioc"/> provides the ability to configure services in a singleton, thread-safe
+    /// service provider instance, which can then be used to resolve service instances.
+    /// The first step to use this feature is to declare some services, for instance:
+    /// <code>
+    /// public interface ILogger
+    /// {
+    ///     void Log(string text);
+    /// }
+    /// </code>
+    /// <code>
+    /// public class ConsoleLogger : ILogger
+    /// {
+    ///     void Log(string text) => Console.WriteLine(text);
+    /// }
+    /// </code>
+    /// Then the services configuration should then be done at startup, by calling one of
+    /// the available <see cref="ConfigureServices(IServiceCollection)"/> overloads, like so:
+    /// <code>
+    /// Ioc.Default.ConfigureServices(collection =>
+    /// {
+    ///     collection.AddSingleton&lt;ILogger, Logger>();
```

Comment:
Is this inconsistent encoding of angle brackets deliberate/necessary?

Reply:
Yup, that's needed to escale the opening bracket, otherwise the parser would consider that as another XML element for that XML docs. Switching to a normal bracket there causes a bunch of errors and squiggly lines to be displayed, see here:

![image](https://user-images.githubusercontent.com/10199417/83448948-0a48b780-a453-11ea-989e-57695d676714.png)

Reply:
@Sergio0694 I think @mrlacey is saying if we do the encoing on the left side, it may just make sense to also do it on the right (even if not technically needed) to just be more consistent?

Reply:
Oh, sorry, read that backwards ahahah
Yeah sure, we can normalize those if you prefer! 👍